views/js/runner/proxy/cache/proxy.js

```diff
@@ -191,33 +158,48 @@ define([
                 var testData    = this.getDataHolder().get('testData');
                 var testContext = this.getDataHolder().get('testContext');
                 var testMap     = this.getDataHolder().get('testMap');
+                var offlinePauseErrorData = offlineErrorHelper.getOfflinePauseError();
 
 
-                if( action === 'pause' ) {
-                    if(actionParams.reason){
-                        offlinePauseError.data = actionParams.reason;
+                if (action === 'pause') {
+                    if (actionParams.reason){
+                        offlinePauseErrorData.data = _.merge(offlinePauseErrorData.data, {
+                            reason: actionParams.reason
+                        });
                     }
-                    throw offlinePauseError;
+
+                    throw _.assign(new Error(offlinePauseErrorData.message), offlinePauseErrorData.data);
```

Comment:
It could be nice to simplify this by using a helper that will construct the error.
Something like:
```javascript
buildErrorFromContext(errorData, context) {
    var err = _.assign(new Error(errorData.message), errorData.data);
    return _.assign(err, context);
}
```
Then:
```javascript
throw buildErrorFromContext(offlinePauseErrorData, {
    reason: actionParams.reason
});
```